common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
+            if (bss_it->fronthaul && bss_it->backhaul) {
+                LOG(ERROR) << "Not supporting hybrid VAP";
+                abort = true;
+                return;
+            }
+
+            // settings
+
+            // ssid
+            conf.set_create_vap_value(vap, "ssid", bss_it->ssid);
+
+            // Hostapd "wpa" field.
+            // This field is a bit field that can be used to enable WPA (IEEE 802.11i/D3.0)
+            // and/or WPA2 (full IEEE 802.11i/RSN):
+            // bit0 = WPA
+            // bit1 = IEEE 802.11i/RSN (WPA2) (dot11RSNAEnabled)
+            int wpa = 0;
+
+            // Set of accepted key management algorithms (WPA-PSK, WPA-EAP, or both). The
+            // entries are separated with a space. WPA-PSK-SHA256 and WPA-EAP-SHA256 can be
+            // added to enable SHA256-based stronger algorithms.
+            // WPA-PSK = WPA-Personal / WPA2-Personal
+            std::string wpa_key_mgmt(""); // default to empty -> delete from hostapd config
+
+            // (dot11RSNAConfigPairwiseCiphersTable)
+            // Pairwise cipher for WPA (v1) (default: TKIP)
+            //  wpa_pairwise=TKIP CCMP
+            // Pairwise cipher for RSN/WPA2 (default: use wpa_pairwise value)
+            //  rsn_pairwise=CCMP
+            std::string wpa_pairwise(""); // default to empty -> delete from hostapd config
+
+            // WPA pre-shared keys for WPA-PSK. This can be either entered as a 256-bit
+            // secret in hex format (64 hex digits), wpa_psk, or as an ASCII passphrase
+            // (8..63 characters), wpa_passphrase.
+            std::string wpa_passphrase("");
+            std::string wpa_psk("");
+
+            // ieee80211w: Whether management frame protection (MFP) is enabled
+            // 0 = disabled (default)
+            // 1 = optional
+            // 2 = required
+            std::string ieee80211w("");
+
+            // This parameter can be used to disable caching of PMKSA created through EAP
+            // authentication. RSN preauthentication may still end up using PMKSA caching if
+            // it is enabled (rsn_preauth=1).
+            // 0 = PMKSA caching enabled (default)
+            // 1 = PMKSA caching disabled
+            std::string disable_pmksa_caching("");
+
+            // Opportunistic Key Caching (aka Proactive Key Caching)
+            // Allow PMK cache to be shared opportunistically among configured interfaces
+            // and BSSes (i.e., all configurations within a single hostapd process).
+            // 0 = disabled (default)
+            // 1 = enabled
+            std::string okc("");
+
+            // This parameter can be used to disable retransmission of EAPOL-Key frames that
+            // are used to install keys (EAPOL-Key message 3/4 and group message 1/2). This
+            // is similar to setting wpa_group_update_count=1 and
+            std::string wpa_disable_eapol_key_retries("");
```

Comment:
Someone, somewhere in the past, decided that we better touch these parameters.
My guess is that we'll need to revisit this code anyhow.
So I am leaving these here until farther investigation and consideration.
My main fear is that by removing them now we'll get into non-working state that it would be hard to deal with.
Now that that the framework for setting parameters's values is set, I prefer having a [different issue](https://jira.prplfoundation.org/browse/PPM-326)  dealing with setting them correctly.